src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import memoize from 'lodash/memoize';
+import Internationalize from 'box-ui-elements/es/elements/common/Internationalize';
+import {
+    readableTimeCellRenderer,
+    sizeCellRenderer,
+    itemNameCellRenderer,
+} from 'box-ui-elements/es/features/virtualized-table-renderers';
+import VirtualizedTable from 'box-ui-elements/es/features/virtualized-table';
+import { Column } from 'react-virtualized/dist/es/Table/index';
+import { TABLE_COLUMNS } from './constants';
+
+const { KEY_NAME, KEY_MODIFIED_AT } = TABLE_COLUMNS;
+
+class ArchiveExplorer extends React.Component {
+    static propTypes = {
+        itemCollection: PropTypes.arrayOf(
+            PropTypes.shape({
+                type: PropTypes.string.isRequired,
+                absolute_path: PropTypes.string.isRequired,
+                name: PropTypes.string.isRequired,
+                modified_at: PropTypes.string.isRequired,
+                size: PropTypes.number.isRequired,
+                path_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string,
+                            name: PropTypes.string,
+                        }),
+                    ),
+                }),
+                parent: PropTypes.string,
+                item_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string.isRequired,
+                            name: PropTypes.string,
+                        }),
+                    ).isRequired,
+                }).isRequired,
+            }),
+        ).isRequired,
+    };
+
+    buildCache = memoize(itemList => {
```

Comment:
Pass in `props` directly so this can be an instance method. There is also naming disparity between `itemList` as a parameter and `itemCollection` in render. Let's keep our naming consistent.